DOCKER.md

```diff
@@ -0,0 +1,83 @@
+#docker-compose run server /bin/bash
+
+We are running three containers 
+One for the front end using react
+One for the back end using rails 
+A postgres data base. 
+
+
+Useful Commands 
+(note when I use the word server, that is just the name of the container with the server running on it, you could sub in the words my-app or postgres)
+
+Creates the containers in detached mode so you get the cursor back. 
+```
+docker-compose up -d
+```
+Removes the containers, also removes the volumes with the added -v. You could add server to the end to only down that container. 
+
+```
+docker-compose down -v
+```
+Useful When Degugging
+If you are trying to debug you can also just run one of the apps with 
+```
+docker-compose up -d server
+```
+You can also use the following to view the logs of a particular container. 
+```
+docker-compose logs server
+```
+If your changes are not taking affect you can use the following to rebuild the image
+```
+docker-compose up --build my-app
+```
+You could also you the following to get rid of the image cache and start from scratch when building the image. 
+```
+docker-compose build --no-cache my-app
+```
+If your container is failing at some point and point, or exiting randomly, you can run which will bring you to the command line of this newly created container. 
+```
+docker-compose run server /bin/sh
+```
+
+Say you just used the standard docker-compose up command, you can use the following commands to enter into the command line of each of the three containers which are now runing. 
+
+```
+```
+```
+```
+Allow you to view command line of postgres container. 
+```
+docker container exec -it postgres bash
+```
+Go to the mentormentee database
+```
+psql -U mentoruser mentormentee
+```
+List databases
+```
+\l
+```
+See tables
+```
+\dt
+```
+Select all items from api_user table (could be any table, you found names using \dt). Don't forget the ;
+```
+SELECT * FROM api_users;
+```
+Quit psql
+```
+\q
+```
+Reminders. 
+1. You can only run one CMD command in your Dockerfile, if you need more commands you need to put them in a bash script, don't forget to add shebang and make it execuable
+I made *init_script.sh* executable by runing. This is part of the repo and you don't need to redo this. 
```

Comment:
update: in docker-compose I added all environment variables for postgres into server too and now I see it running. I found to do that when I saw database.yml uses it but on the server image these environment variables are not declared so db:migrate was not running.

Reply:
Great to hear you were able to get it working! Where in the docker-compose.yml did you add the environment variables? You were saying you added them under server? perhaps you can show what your docker-compose.yml looks like now?

Reply:
here it is
![image](https://user-images.githubusercontent.com/13537860/91587085-92593300-e956-11ea-82b8-82811611f7d3.png)

btw, it is really quick and painless to start working now, thanks for making this happen!

Reply:
Thanks for trying it out on Windows and adding the notes for doing so! Glad it works now! I'll be having a look at some of the other comments below.

Reply:
I decided to try developing a bit on the front-end and the code was not refreshed for me during development. I have the changes I made that work for me in branch 13-landing-page, along with the environment variables above in another commit. Feel free to cherry-pick them and them to this branch here, if you'd like, otherwise, I will add them with my PR.

here are the exact cherry-picks
```
git cherry-pick a9c765dd51caa3ba2f5f29425216d60d15e7f027

git cherry-pick e048b083d8abd36ac0c40fd1f5abd744255bea0d
```

Reply:
added in the environment variables, and made the secret a .env variable

Reply:
@diwanow I didn't do the cherrypicking add, I might give it a go later but I haven't used cherrypick often. If you can add them in your PR that would be great.